imports/client/components/AccountForm.tsx

```diff
@@ -427,51 +498,122 @@ const AccountForm = (props: AccountFormProps) => {
             successMessage ? (
               <Alert variant="success">{successMessage}</Alert>
             ) : null}
+            {props.format === AccountFormFormat.INVITATION_WELCOME ? (
+              <div>
+                <p>
+                  <b>New user?</b>{" "}
+                  <NoPaddingLinkButton
+                    variant="link"
+                    onClick={onInvitationNewUserSelected}
+                  >
+                    Create an account
+                  </NoPaddingLinkButton>
+                  .
+                </p>
+                <p>
+                  <b>Existing user?</b>{" "}
+                  <NoPaddingLinkButton
+                    variant="link"
+                    onClick={onInvitationExistingUserSelected}
+                  >
+                    Sign in
+                  </NoPaddingLinkButton>
+                  .
+                </p>
+              </div>
+            ) : null}
             {props.format === AccountFormFormat.LOGIN ||
-            props.format === AccountFormFormat.REQUEST_PW_RESET
+            props.format === AccountFormFormat.REQUEST_PW_RESET ||
+            (props.format === AccountFormFormat.ENROLL &&
+              props.huntInvitationCode &&
+              googleSignInCredentials === undefined)
               ? emailInput
               : null}
             {props.format === AccountFormFormat.LOGIN ||
-            props.format === AccountFormFormat.ENROLL ||
+            (props.format === AccountFormFormat.ENROLL &&
+              googleSignInCredentials === undefined) ||
             props.format === AccountFormFormat.RESET_PWD
               ? pwInput
               : null}
+            {props.format === AccountFormFormat.ENROLL &&
+            googleSignInCredentials === undefined ? (
+              <div className="d-grid gap-2 mt-3 justify-content-center">
+                <div>
+                  or{" "}
+                  <SignInWithGoogleButton
+                    submitting={submitting}
+                    onStarted={() => {
+                      setSubmitState(AccountFormSubmitState.SUBMITTING);
+                    }}
+                    onSucceeded={(token: string, secret: string) => {
+                      setSubmitState(AccountFormSubmitState.IDLE);
+                      setGoogleSignInCredentials({
+                        token,
+                        secret,
+                      });
+                    }}
+                    onCanceled={() => {
+                      setSubmitState(AccountFormSubmitState.IDLE);
+                    }}
```

Comment:
FTR, our strategy has generally been to not put anonymous functions directly in JSX code, and instead wrap everything with `useCallback`. My understanding is that otherwise, anonymous functions will get regenerated on every render of this component, which could trigger spurious re-renders of the components that are receiving those functions as props.

It's also the case that we may be overly concerned about that, but I think my preference is to stick with it as a style preference across the codebase.

Reply:
Sure, done.